Pod/Classes/PINRemoteImageManager.m

```diff
@@ -1058,6 +1114,16 @@ - (void)didReceiveData:(NSData *)data forTask:(NSURLSessionDataTask *)dataTask
                 task.progressImage.progressThresholds = self.progressThresholds;
             }
         }
+    
+        for (PINRemoteImageCallbacks *completion in task.callbackBlocks.allValues) {
```

Comment:
I think it might be better to copy the completions and call them outside of the lock?